### PR TITLE
TC_02.04.023 | Dashboard > Command panel on the left side

### PR DESCRIPTION
**Pull request by Kamal-Zh**
https://trello.com/c/GwDMWKg0/232-tc0204023-dashboard-command-panel-on-the-left-side

### CODE REVIEW

**Comment by Rom-T** [comment]
A very difficult test to read.  Try using a for loop to go through all the elements.

File: cypress/e2e/Project/dashboardSideMenu.cy.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+///<reference types ="cypress"/>
+
+describe('sideMenu', () => {
+
+
+    it('"Dashboard > Side menu"', function () {
+
+    //New Item
```
Comment (KravchukKseniya) [inline cypress/e2e/Project/dashboardSideMenu.cy.js:8]: Avoid comments in your code

File: cypress/e2e/Project/dashboardSideMenu.cy.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    cy.get('[href="/view/all/newJob"]')
+    .invoke('text')
+    .then((text) => {
+        const trimmedNewItem = text.trim();
+        expect(trimmedNewItem).to.equal('New Item');
+    })
+    cy.get('[href="/view/all/newJob"]').click();
+    cy.url().should('be.equal', 'http://localhost:8080/view/all/newJob');
```
Comment (KravchukKseniya) [inline cypress/e2e/Project/dashboardSideMenu.cy.js:17]: Please, do not use this hard code links, check only end points `view/all/newJob`. We discussed that on the last lesson with Maria.
You can use `.should('contain', '/view/all/newJob')` for this purpose.

File: cypress/e2e/Project/dashboardSideMenu.cy.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    .invoke('text')
+    .then((text) => {
+        const cleanedTextMyViews = text.trim();
+        expect(cleanedTextMyViews).to.equal('My Views');
+    })
+    cy.get('[href="/me/my-views"]').click();
+    cy.url().should('be.equal', 'http://localhost:8080/me/my-views/view/all/');
+    cy.get('[id="jenkins-home-link"]').should('be.visible').click()
+    
+    });
```
Comment (KravchukKseniya) [inline cypress/e2e/Project/dashboardSideMenu.cy.js:68]: You need to divide your test case to several test cases. There are a lot of assertions in one test.

At first check that left-side menu have correct text. This code form our first review with Stan can be helpful
https://redroverschool.slack.com/archives/C01DNKMFN73/p1697513993568089?thread_ts=1697513945.186669&cid=C01DNKMFN73

Than check work of every link in separate test.

Also, you can move all data like menu item names and endpoints to fixture file.

**Review by KravchukKseniya — CHANGES_REQUESTED** [review]

File: cypress/e2e/Project/dashboardSideMenu.cy.js (hunk 1 above)
Comment (Rom-T) [reply cypress/e2e/Project/dashboardSideMenu.cy.js:8]: Don't neglect Request Changes due to minor errors.

**Comment by Paul-Korz** [comment]
Hi. Please close your PR and delete branch.

**Pull request closed without merging**